website/static/js/paginator.js

```diff
@@ -111,12 +111,16 @@ var Paginator = oop.defclass({
         }
     },
     nextPage: function(){
-        this.currentPage(this.currentPage() + 1);
-        this.fetchResults();
+        if (this.currentPage() !== this.numberOfPages() - 1) {
```

Comment:
Why isn't the button disabled when you click the first time? If it were tied to `currentPage`, then that gets incremented before the request returns, so the button should disable.